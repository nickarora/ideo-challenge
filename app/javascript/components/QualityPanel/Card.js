import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Container = styled.article`
  border-radius: 5px;
  border-style: solid;
  border-width: 5px;
  box-shadow: 0px 1px 24px 0px rgba(0, 0, 0, 0.33);
  margin-bottom: 21px;
  text-align: center;
`

const Card = ({ quality, children }) => (
  <Container
    style={{
      borderColor: quality.color,
    }}
  >
    {children}
  </Container>
)

Card.propTypes = {
  quality: PropTypes.shape({
    color: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.node.isRequired,
}

export default Card
