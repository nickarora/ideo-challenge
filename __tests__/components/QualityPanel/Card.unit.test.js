import Card from '~/components/QualityPanel/Card'

describe('Card', () => {
  it('renders children with a given color', () => {
    const wrapper = shallow(
      <Card
        quality={{
          color: '#333',
        }}
      >
        CONTENT
      </Card>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
