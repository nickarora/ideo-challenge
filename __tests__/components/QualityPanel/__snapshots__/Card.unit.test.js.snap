// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders children with a given color 1`] = `
<styled.article
  style={
    Object {
      "borderColor": "#333",
    }
  }
>
  CONTENT
</styled.article>
`;
